### PR TITLE
Create Approvals test doc

### PR DESCRIPTION
**Pull request by maestromac**
## What type of PR is this? (check all applicable)
- [x] Documentation

## Description
Create a new doc

## Related Tickets & Documents
n/a

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
n/a

## Added tests?
- [x] no, because they aren't needed

## Added to documentation?
- [x] docs.dev.to

### CODE REVIEW

File: docs/tests/approvals-tests.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+---
+title: Approvals Tests
+---
+
+# Approvals Tests
+
+Approvals are based on the idea of the golden master. You take a snapshot of an
```
Comment (Ridhwana) [inline docs/tests/approvals-tests.md]: I think a link to a something that explains the Golden Master will be very beneficial here. I don't think everyone will know what it is.

Comment (Ridhwana) [inline docs/tests/approvals-tests.md]: I feel like the sentence 'You take a snapshot' is not very articulate here. I would be more specific if possible and replace you with specificity.

File: docs/tests/approvals-tests.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+object and then compare all future versions of the object to the snapshot. A
+good use-case for Approvals tests is comparing rendered HTML.
```
Comment (Ridhwana) [inline docs/tests/approvals-tests.md]:
```suggestion
good use-case for the Approvals tests is to compare the rendered HTML to the test specified HTML.
```

I feel like when using the word compared you usually need to specify what to its being compared to.

File: docs/tests/approvals-tests.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+## Usage
+
+Simply create your test similar to the following
```
Comment (Ridhwana) [inline docs/tests/approvals-tests.md]:
```suggestion
Simply create your test similar to the following:
```

File: docs/tests/approvals-tests.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+```ruby
+it "works" do
```
Comment (Ridhwana) [inline docs/tests/approvals-tests.md]:
```suggestion
it "renders the correct HTML on the home page" do
```
I feel like a more realistic test description will be more beneficial

File: docs/tests/approvals-tests.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  page = '<html><head></head><body><h1>ZOMG</h1></body></html>'
+  verify(format: :html) { page } # format can also be :json
+end
+```
+
+then run the said test and a new `*.received.*` file will be created. You will
```
Comment (Ridhwana) [inline docs/tests/approvals-tests.md]:
```suggestion
You may then run the said test and a new `*.received.*` file will be created. Thereafter,
```

File: docs/tests/approvals-tests.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+then run the following to verify the newly created file and approve the created
```
Comment (Ridhwana) [inline docs/tests/approvals-tests.md]:
```suggestion
run the following command to verify the newly created file and approve the created
```

File: docs/tests/approvals-tests.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+```shell
+approvals verify
+```
+
+Please be sure to include the Approvals file in your commit.
+
+## Edge cases
+
+Approvals test isn't great for testing changing variables. That includes
```
Comment (Ridhwana) [inline docs/tests/approvals-tests.md]:
```suggestion
Approvals tests are difficult to utilise for testing variables that change. This includes variables like:
```

File: docs/tests/approvals-tests.md (hunk 1 above)
Comment (maestromac) [reply docs/tests/approvals-tests.md]: I quite literally took this from their [official repo](https://github.com/kytrinyx/approvals). Yeah, I agree that pluralizing it sounds a bit off but I'm just keeping it consistent with the gem name.

Comment (maestromac) [reply docs/tests/approvals-tests.md]: I think I'm going to go ahead and remove that as it's not really necessary.

**Comment by maestromac** [comment]
Thank you for the thorough review @Ridhwana !
